pkg/build/controller/buildpod/factory.go

```diff
@@ -0,0 +1,213 @@
+package buildpod
```

Comment:
to keep the object/event processing code separate from the setup/creation code. We don't need to call it factory ... maybe infra.go ? The deployment controllers still call it factory, which is why I reused the name.